src/careamics/config/training_model.py

```diff
@@ -37,6 +37,20 @@ class TrainingConfig(BaseModel):
     num_epochs: int = Field(default=20, ge=1)
     """Number of epochs, greater than 0."""
 
+    precision: Literal["64", "32", "16-mixed", "bf16-mixed"] = Field(default="32")
+    """Numerical precision"""
+    max_steps: int = Field(default=-1, ge=-1, exclude=0)
+    """Maximum number of steps to train for. -1 means no limit."""
+    check_val_every_n_epoch: Optional[int] = Field(default=1, ge=1)
+    """Validation step frequency."""
+    enable_progress_bar: Optional[bool] = Field(default=True)
```

Comment:
Why are `check_val_every_n_epoch` and `enable_progress_bar` optional? Does setting `None` has a meaning here?

I see that they are so in the PL Trainer, but in our case since they have a default value, they will never be `None`.